### ISSUE TITLE
Uncaught TypeError: material.onBeforeRender is not a function

### ISSUE THREAD

**Issue by Enijar** [felixmariotto/three-mesh-ui#149]
The error below is thrown when using a version of `three` >= `0.133.0`

```text
Uncaught TypeError: material.onBeforeRender is not a function
```

Minimal reproducible example [CSB](https://codesandbox.io/s/damp-haze-uqme4?file=/index.html):

```html
<script type="module">
      import * as THREE from "https://cdn.jsdelivr.net/npm/three@0.137.1/build/three.module.js";
      import { OrbitControls } from "https://cdn.jsdelivr.net/npm/three@0.121.1/examples/jsm/controls/OrbitControls.js";
      import "https://cdn.jsdelivr.net/npm/three-mesh-ui@6.2.0/build/three-mesh-ui.js";

      const renderer = new THREE.WebGLRenderer();
      renderer.setSize(window.innerWidth, window.innerHeight);
      document.body.appendChild(renderer.domElement);

      const aspect = window.innerWidth / window.innerHeight;
      const camera = new THREE.PerspectiveCamera(75, aspect, 0.1, 1000);
      const controls = new OrbitControls(camera, renderer.domElement);

      const scene = new THREE.Scene();

      camera.position.z = 5;

      const panel = new ThreeMeshUI.Block({
        width: 1,
        height: 0.8,
        fontSize: 0.055,
        justifyContent: "center",
        alignContent: "center"
      });

      scene.add(panel);

      (function animate() {
        // @note to prevent errors being spammed
        // requestAnimationFrame(animate);

        renderer.setSize(window.innerWidth, window.innerHeight);
        camera.aspect = window.innerWidth / window.innerHeight;
        camera.updateProjectionMatrix();

        ThreeMeshUI.update();
        controls.update();
        renderer.render(scene, camera);
      })();
    </script>
```

**Comment by swingingtom** [comment]
Hey @Enijar,

Thanks for the sandbox. May I ask if you already succeed to make that kind of setup work using older versions? I really used to jsm libraries and I didn't even know we were able to load a **non jsm** file through `import` ...

```javascript 
import "https://cdn.jsdelivr.net/npm/three-mesh-ui@6.2.0/build/three-mesh-ui.js";
```

My guess here is the **js** build of three-mesh-ui ( which also package threejs without globalifying it ) conflict with THREE import. Such as described [here](https://stackoverflow.com/questions/69817153/three-js-typeerror-material-onbeforerender-is-not-a-function)
> I highly suspect that the runtime error is produced by the fact that you use library files from different releases. Please try it again with the latest release r134. – 
Mugen87 Nov 3 '21 at 8:07

Fortunately the **three-mesh-ui 6.2.0 jsm** build, seems globally compatible with three r137.1 ([here](https://codesandbox.io/s/tmu-properties-overview-zeggq?file=/src/sandbox.js)) 

Did you need to rely on the js build of three-mesh-ui? Or did you do so because there is no cdn of three-mesh-ui jsm?

**Comment by Enijar** [comment]
>May I ask if you already succeed to make that kind of setup work using older versions?

It works on 5 minor versions < `0.133.0` (that I've tested). I only use the `umd` imports to keep the reproducible demo small and portable. I've tested this using my own build process, which uses `npm` and `webpack`, and has the same error for versions of `three` < `0.133.0`.

Let me test with the `jsm` build and see if the error still happens. The reason I'm doing this is to test my TypeScript PR for this library [WIP PR here](https://github.com/felixmariotto/three-mesh-ui/pull/146), which is why I came across this issue in the first place.
